src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Util.java

```diff
@@ -46,37 +47,37 @@ public final class P11Util {
 
     // Used by PBE
     static final class KDFData {
-        public enum Operation {ENCRYPTION, AUTHENTICATION, GENERIC}
-        public long kdfMech;
-        public long prfMech;
-        public String keyAlgo;
-        public int keyLen;
-        public Operation op;
+        public final long kdfMech;
+        public final long prfMech;
+        public final String keyAlgo;
+        public final int keyLen;
+        public final CK_ATTRIBUTE encrypt_or_sign_true;
+
         KDFData(long kdfMech, long prfMech, String keyAlgo,
-                int keyLen, Operation op) {
+                int keyLen, CK_ATTRIBUTE encrypt_or_sign_true) {
             this.kdfMech = kdfMech;
             this.prfMech = prfMech;
             this.keyAlgo = keyAlgo;
             this.keyLen = keyLen;
-            this.op = op;
+            this.encrypt_or_sign_true = encrypt_or_sign_true;
         }
 
         public static void addPbkdf2Data(String algo, long kdfMech,
                                          long prfMech) {
             kdfDataMap.put(algo, new KDFData(kdfMech, prfMech,
-                    "Generic", -1, Operation.GENERIC));
+                    algo, -1, CK_ATTRIBUTE.SIGN_TRUE));
         }
 
         public static void addPbkdf2AesData(String algo, long kdfMech,
                                             long prfMech, int keyLen) {
             kdfDataMap.put(algo, new KDFData(kdfMech, prfMech,
-                    "AES", keyLen, Operation.ENCRYPTION));
+                    "AES", keyLen, CK_ATTRIBUTE.ENCRYPT_TRUE));
         }
 
-        public static void addPkcs12KDData(String algo, long kdfMech,
-                                         int keyLen) {
+        public static void addPkcs12KDMacData(String algo, long kdfMech,
+                                              int keyLen) {
             kdfDataMap.put(algo, new KDFData(kdfMech, -1,
-                    "Generic", keyLen, Operation.AUTHENTICATION));
+                    "HMAC", keyLen, CK_ATTRIBUTE.SIGN_TRUE));
```

Comment:
Can you please elaborate a bit on the rationale and consequences of replacing "Generic" with "HMAC"?